### DISCUSSION TITLE
using "console --picture ..." without imgfree aferward , causes Initramfs unpacking failed: invalid magic at kernel start

### DISCUSSION THREAD

**Discussion by ptempier** [ipxe/ipxe#1263] [General]
I Can't understand the relation between :
using a background image in ipxe menu
imgfree which discard kernel images and not pictures
having the kernel boot which should results in wiping anything set in the ipxe context except cmdline parameter
and then getting  Initramfs unpacking failed: invalid magic at start of compressed archive

Maybe not a bug but ... i just can't find the relation.

**Comment by NiKiZe** [comment]
Depending on how your script works, yes that is possible due to the load protocol.
Could you use imgstat before your boot command and post the output and maybe we can explain what is going on in your case.

**Reply by ptempier** [reply]
Hello it says :
spash.png 24064 bytes [png]
vmlinuz : 14966600 bytes [efi][selected]
initrd : 149398652 bytes

also typing imgfree and imgstat afterward, shows that it discarded both the pictrue and the kernel/initrd

**Reply by mcb30** [reply]
> Hello it says :
> spash.png 24064 bytes [png]
> vmlinuz : 14966600 bytes [efi][selected]
> initrd : 149398652 bytes
> 
> also typing imgfree and imgstat afterward, shows that it discarded both the pictrue and the kernel/initrd

You have presumably used the `--keep` option to the `console` command to prevent it from automatically deregistering the `splash.png` image. 

Remove the `--keep` option and everything should work as expected.

**Reply by NiKiZe** [reply]
Another workaround is to use imgfree before your initrd and kernel lines.

**Reply by ptempier** [reply]
I have the workaround , what i dont understand is why.
If i dont use imgfree, does ipxe  tries to unpack the png in place of the initrd thus i get the invalid magic ?
But in imgstats ipxe knows that one file is of type png

**Reply by NiKiZe** [reply]
iPXE? No!
Your kernel, yes maybe.
